clippy_lints/src/utils/author.rs

```diff
@@ -617,9 +617,13 @@ impl<'tcx> Visitor<'tcx> for PrintVisitor {
                     start_pat, end_pat, end_kind, current
                 );
                 self.current = start_pat;
-                self.visit_expr(start);
+                if let Some(expr) = start {
+                    self.visit_expr(expr);
```

Comment:
btw, there's a nice `syntax::walk_list!(self, visit_expr, start)` pattern for this sorta thing.